### PR TITLE
cache reader should turn back values that were assoc array to assoc a…

### LINKED ISSUE #21896: Sharing with LDAP Users not working on subsequent attempts when cache is enabled
### Steps to reproduce
1. Having cache enabled: APCu 4.0.7, Redis 2.2.7 
2. Trying to share some files/folders with an LDAP User
2. When typing the first letters of the Username, User is being correctly found on the first attempt
3. However, when deleting the letters and searching again for the same User, this cannot be found and the search seems to be stuck
4. This only happens when cache is enabled

### Expected behaviour

LDAP User should also be found on the second attempt

### Actual behaviour

See above

### Server configuration
**Operating system**:

Debian Jessie

**Web server:**

Apache 2.4

**Database:**

Percona XtraDB 5.6.27

**PHP version:**

5.6

**ownCloud version:** (see ownCloud admin page)

8.2.2

**Updated from an older ownCloud or fresh install:**

Fresh install

**Are you using external storage, if yes which one:** local/smb/sftp/...

local

**Are you using encryption:** yes/no

no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

LDAP

@MorrisJobke can you maybe help here?

00004656

### PR DESCRIPTION
**Pull request by blizzz**
…rrays again

Fixes #21896 (see for reproduction steps there as well)

please review @owncloud/ldap 

@karlitschek ok to backport?

### CODE REVIEW

File: apps/user_ldap/lib/connection.php [hunk 1]
```diff
@@ -206,7 +206,13 @@ public function getFromCache($key) {
 		}
 		$key = $this->getCacheKey($key);
 
-		return json_decode(base64_decode($this->cache->get($key)));
```
Comment (LukasReschke) [inline apps/user_ldap/lib/connection.php]: Tried the assoc parameter of json_decode?

Comment (MorrisJobke) [inline apps/user_ldap/lib/connection.php]: https://github.com/owncloud/core/pull/21999

Comment (blizzz) [inline apps/user_ldap/lib/connection.php]: Subject: Re: [core] cache reader should turn back values that were assoc
 array to assoc =?UTF-8?B?YeKApg==?= (#21998)
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: base64
Date: Fri, 29 Jan 2016 09:50:24 +0000
MIME-Version: 1.0


Tm8sIGRpZCBub3QsIGlnbm9yZWQgdGhhdCBvbmUgd2hpbGUgYmVpbmcgdW5oYXBweSBub3RoaW5n
IGFkZXF1YXRlIHdhcyBhdmFpbGFibGUgd2l0aCB0aGUgICRvcHRpb25zIHBhcmFtZXRlci4uLg0K
DQpPbiBGcmkgSmFuIDI5IDA0OjIxOjI4IDIwMTYgR01UKzAxMDAsIEx1a2FzIFJlc2Noa2Ugd3Jv
dGU6DQo+IEluIGFwcHMvdXNlcl9sZGFwL2xpYi9jb25uZWN0aW9uLnBocDoNCj4gPiBAQCAtMjA2
LDcgKzIwNiwxMyBAQCBwdWJsaWMgZnVuY3Rpb24gZ2V0RnJvbUNhY2hlKCRrZXkpIHsNCj4gPiAg
CQl9DQo+ID4gIAkJJGtleSA9ICR0aGlzLT5nZXRDYWNoZUtleSgka2V5KTsNCj4gPiAgDQo+ID4g
LQkJcmV0dXJuIGpzb25fZGVjb2RlKGJhc2U2NF9kZWNvZGUoJHRoaXMtPmNhY2hlLT5nZXQoJGtl
eSkpKTsNCj4gVHJpZWQgdGhlIGFzc29jIHBhcmFtZXRlciBvZiBqc29uX2RlY29kZT8gDQo+IOKA
lA0KPiBSZXBseSB0byB0aGlzIGVtYWlsIGRpcmVjdGx5IG9yIHZpZXcgaXQgb24gR2l0SHViLu+/
vA0KPg0KDQotLSANClNlbnQgZnJvbSBteSBKb2xsYQ==

**Comment by blizzz** [comment]
Closed in favor of #21999

**Pull request closed without merging**